docs/integration-guide.md

```diff
@@ -107,6 +107,319 @@ Here's how we recommend doing it:
+
+* `templates/zerver/integrations.html`: Edit to add end-user documentation and
+  integration icon. See [Documenting your
+  integration](#documenting-your-integration) for details.
+* `zerver/test_hooks.py`: Edit to include tests for your webbook. See [Testing
+  and writing tests](testing.html) for details.
+* `zproject/urls.py`: Edit to add externally available url of the webhook and
+  associate with the function added to `zerver/views/webhooks/mywebhook.py`
+
+### Files that need to be created
+
+Select a name for your webhook and use it consistently. The examples below are
+for a webhook named 'mywebhook'.
+
+* `static/images/integrations/logos/mywebhook.png`: An image to represent
+  your integration in the user interface. Generally this Should be the logo of the
+  platform/server/product you are integrating. See [Documenting your
+  integration](#documenting-your-integration) for details.
+* `static/images/integrations/mywebbook/001.png`: A screen capture of your
+  integration for use in the user interface. You can add as many images as needed
+  to effectively document your webhook integration. See [Documenting your
+  integration](#documenting-your-integration) for details.
+* `zerver/fixtures/mywebhook/mywebhook_messagetype.json`: Sample json payload data
+  used by tests. Add one fixture file per type of message supported by your
+  integration. See [Testing and writing tests](testing.html) for details.
+* `zerver/views/webhooks/mywebhook.py`: Includes the main webhook integration
+  function including any needed helper functions.
+
+### Walkthrough of `Hello World` webhook
+
+Below explains each part of a simple webhook integration, called **Hello
+World**. This webhook sends a "hello" message to the `test` stream and includes
+a link to the Wikipedia article of the day, which it formats from json data it
+receives in the http request.
+
+Use this walkthrough to learn how to write your first webhook
+integration.
+
+#### Step 0: Create fixtures
+
+The first step in creating a webhook is to examine the data that the
+service you want to integrate will be sending to Zulip.
+
+You can use [requestb.in](http://requestb.in/) or a similar tool to capture
+webook payload(s) from the service you are integrating. Examining this
+data allows you to do two things:
+
+1. Determine how you will need to structure your webook code, including what
+   message types your integration should support and how; and,
+2. Create fixtures for your webook tests.
+
+Fixtures enable the testing of webhook integration code without the need to
+actually contact the service being integrated.
+
+Because `Hello World` is a very simple webhook that does one thing, it requires
+only one fixture, `zerver/fixtures/helloworld/helloworld_hello.json`:
+
+```
+{
+  "featured_title":"Marilyn Monroe",
+  "featured_url":"https://en.wikipedia.org/wiki/Marilyn_Monroe",
+}
+```
+
+When writing your own webhook integration, you'll want to write one test
```

Comment:
`at least one test`?

Message type shouldn't determine the number of tests I think. It should rather depend on conditions ( I met some situations like this)